table/block_builder.cc

```diff
@@ -124,7 +157,18 @@ void BlockBuilder::Add(const Slice& key, const Slice& value) {
   buffer_.append(key.data() + shared, non_shared);
   buffer_.append(value.data(), value.size());
 
+  if (data_block_hash_index_builder_) {
+    // two largest numbers for uint8_t is used as speical flags
+    // const uint8_t kNoEntry = 255;
+    // const uint8_t kCollision = 254;
+    // so normal restart index cannot be these values.
+    assert(restarts_.size() < kCollision);
```

Comment:
If I understand correctly: max number of restarts we now generally support is 2^31 - 1, and in hash index case it will be 253. Is that right?

Reply:
Yes, that's right. Let me include it in the code comment.